### PR TITLE
util/cbfstool/flashmap/fmap.c: Fix fmaptool endianness bugs on BE

### PR DESCRIPTION
**Pull request by Sakoram**

### CODE REVIEW

File: util/cbfstool/flashmap/fmap.c [hunk 1]
```diff
@@ -48,7 +48,7 @@ static int is_valid_fmap(const struct fmap *fmap)
 		return 0;
 	/* a basic consistency check: flash should be larger than fmap */
 	if (fmap->size <
```
Comment (krystian-hebel) [inline util/cbfstool/flashmap/fmap.c]:
```suggestion
	if (le32toh(fmap->size) <
```

File: util/cbfstool/flashmap/fmap.c [hunk 2]
```diff
@@ -184,7 +184,7 @@ int fmap_print(const struct fmap *fmap)
 	kv_pair_print(kv);
```
Comment (krystian-hebel) [inline util/cbfstool/flashmap/fmap.c:184]: Can't mark proper lines in the review, but just above this there are multi-byte fields printed, they should have their byte orders reversed (base, size and nareas).

Comment (krystian-hebel) [reply util/cbfstool/flashmap/fmap.c:184]: The same goes for lines 196-203 for offset, size and flags, and 206 where the flags are read again.

File: util/cbfstool/flashmap/fmap.c [hunk 3]
```diff
@@ -319,7 +319,7 @@ const struct fmap_area *fmap_find_area(const struct fmap *fmap,
 	if (!fmap || !name)
 		return NULL;
 
-	for (i = 0; i < fmap->nareas; i++) {
+	for (i = 0; i < le16toh((fmap)->nareas); i++) {
```
Comment (krystian-hebel) [inline util/cbfstool/flashmap/fmap.c]:
```suggestion
	for (i = 0; i < le16toh(fmap->nareas); i++) {
```

File: util/cbfstool/flashmap/fmap.c (hunk 2 above)
Comment (krystian-hebel) [reply util/cbfstool/flashmap/fmap.c:184]: And 361, 366, 376 for base, size, nareas (the proper way would be to convert the left side, with `fmap->`, I think).

File: util/cbfstool/flashmap/fmap.c [hunk 4]
```diff
@@ -428,13 +428,13 @@ static int fmap_append_area_test(struct fmap **fmap)
 		goto fmap_append_area_test_exit;
```
Comment (krystian-hebel) [inline util/cbfstool/flashmap/fmap.c:428]: Can't point to the proper lines again: 417-420 should use `htoleXX` on the right side of `=`, except for `name`

File: util/cbfstool/flashmap/fmap.c [hunk 5]
```diff
@@ -428,13 +428,13 @@ static int fmap_append_area_test(struct fmap **fmap)
...
 	(*fmap)->nareas = ~(0);
 	if (fmap_append_area(fmap, 0, 0, (const uint8_t *)"foo", 0) >= 0) {
 		printf("FAILURE: failed to abort with too many areas\n");
 		goto fmap_append_area_test_exit;
 	}
-	(*fmap)->nareas = nareas_orig;
+	(*fmap)->nareas = htole16(nareas_orig);
 
 	total_size = sizeof(**fmap) + sizeof(test_area);
 	if (fmap_append_area(fmap,
```
Comment (krystian-hebel) [inline util/cbfstool/flashmap/fmap.c:440]: Offset, size and flags below should have `leXXtoh`

**Review by krystian-hebel — CHANGES_REQUESTED** [review]

**Comment by miczyg1** [comment]
@krystian-hebel @Sakoram please clean up the PR so that we can merge it

File: util/cbfstool/flashmap/fmap.c (hunk 4 above)
Comment (Sakoram) [reply util/cbfstool/flashmap/fmap.c:428]: I'm not sure if this is really necessary. Those values are only used in lines 441-444, If I use htoleXX here, I would have to use leXXtoh in 441-444. This seems pointless.

**Comment by Sakoram** [comment]
@miczyg1 I fixed everything @krystian-hebel  commented about, except one function. I commented why here: https://github.com/3mdeb/coreboot/pull/21#discussion_r535945841

**Comment by Sakoram** [comment]
@miczyg1  fixed everything @krystian-hebel  commented about, except one function. I commented why here: https://github.com/3mdeb/coreboot/pull/21#discussion_r535945841

File: util/cbfstool/flashmap/fmap.c (hunk 4 above)
Comment (krystian-hebel) [reply util/cbfstool/flashmap/fmap.c:428]: It has to be swapped twice. If someone were to attach a debugger to inspect the memory, he/she should be able to see exactly the same format as if it were written to a file between these two points. Also, these tests may change at some point, someone may add a detailed dump or actually write this data to a file. It is easier to fix the code now than to debug it later.

Comment (Sakoram) [reply util/cbfstool/flashmap/fmap.c:428]: Ok. Corrected.

File: util/cbfstool/flashmap/fmap.c (hunk 5 above)
Comment (krystian-hebel) [reply util/cbfstool/flashmap/fmap.c:440]: Fixed

**Review by krystian-hebel — APPROVED** [review]

**Pull request merged**